src/main/java/com/squareup/javapoet/AnnotationSpec.java

```diff
@@ -98,6 +104,59 @@ private void emitAnnotationValues(CodeWriter codeWriter, String whitespace,
     codeWriter.emit(whitespace + "}");
   }
 
+  public static AnnotationSpec get(Annotation annotation) {
+    return get(annotation, false);
+  }
+
+  public static AnnotationSpec get(Annotation annotation, boolean includeDefaultValues) {
+    Builder builder = builder(annotation.annotationType());
+    try {
+      Method[] methods = annotation.annotationType().getDeclaredMethods();
+      Arrays.sort(methods, new Comparator<Method>() {
+        @Override
+        public int compare(Method m1, Method m2) {
+          return m1.getName().compareTo(m2.getName());
+        }
+      });
+      for (Method method : methods) {
+        Object value = method.invoke(annotation);
+        if (!includeDefaultValues) {
+          if (Objects.deepEquals(value, method.getDefaultValue())) {
+            continue;
+          }
+        }
+        if (value.getClass().isArray()) {
+          for (int i = 0; i < Array.getLength(value); i++) {
+            addAnnotationValue(builder, method.getName(), Array.get(value, i));
+          }
+          continue;
+        }
+        if (value instanceof Annotation) {
+          AnnotationSpec nested = get((Annotation) value);
+          if (nested.members.isEmpty()) {
+            builder.addMember(method.getName(), "@$T", nested.type);
+          } else {
+            String s = nested.toString();
```

Comment:
Better use internal `emitAnnotationValues(...)` here or is grabbing the substring between `(...)` ok?
I had to split the nested toString result from the nested type, because otherwise the import mechanic does not kick in.